### PR TITLE
Upgrade to airflow 2.7.2

### PR DESCRIPTION
**Pull request by Lee-W**
## Types of changes

- **New feature**

## Description
* Upgrade python to 3.10 and airflow to 2.7.2
* Use `airflow standalone` to run airflow components instead of using the shell script logic as we use to do
* we'll need to change the airflow.cfg file as many configuration has been changed during the past few years
* this is an early stage draft

## Checklist:
- [ ] Add test cases to all the changes you introduce
- [ ] Run `poetry run pytest` locally to ensure all linter checks pass
- [ ] Update the documentation if necessary

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
 COPY poetry.lock poetry.lock
 
-RUN python -m poetry install --no-interaction --no-ansi --no-dev \
-    # Cleaning poetry installation's cache for production:
-    && rm -rf "$POETRY_CACHE_DIR" \
-    && pip uninstall -yq poetry
+RUN pip install --no-cache-dir -U pip poetry==1.6.1 \
+    && python -m poetry export -o requirements.txt \
+    && python -m pip uninstall poetry -y \
+    && pip install -r requirements.txt
```
Comment (uranusjr) [inline Dockerfile:22]: Use the official constraints here?

https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html

Comment (Lee-W) [reply Dockerfile:22]: hmmm... i think we want to manage it through poetry which does not support constraints

Comment (Lee-W) [reply Dockerfile:22]: another way might be to remove airflow from requirements.txt and install the constraint file separately. not sure whether we want to do it though

**Comment by Lee-W** [comment]
The last I heard for @Cliying94 . We decide not to upgrade to 2.x. Close this one

**Pull request closed without merging**